### PR TITLE
Feat/6373 example gold workload

### PR DESCRIPTION
**Pull request by james-horrocks**
#### 📲 What

Add an example gold layer workload using the previous silver layer examples

#### 🤔 Why

So that the platform contains working examples of all stages of a medallion architecture data lake

#### 🛠 How

`gold_movies_example` added as a pyspark workload

#### 👀 Evidence

Documentation not yet updated
Successful run in NonProd: https://adb-854800919001492.12.azuredatabricks.net/?o=854800919001492#job/1069938958776254/run/799189810261888

#### 🕵️ How to test

Unit tests can be run using `make test` or `poetry run python -m pytest de_workloads/processing/gold_movies_example/tests/unit`
The workload can be tested in ADF in NonProd

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [x] Documentation has been updated to reflect the changes?
- [x] Passing all automated tests, including a successful deployment?
- [x] Passing any exploratory testing?
- [x] Rebased/merged with latest changes from development and re-tested?
- [x] Meeting the Coding Standards?

### CODE REVIEW

File: de_templates/processing/processing_template/data_factory/pipelines/arm_template.json.jinja
Comment (james-horrocks) [inline de_templates/processing/processing_template/data_factory/pipelines/arm_template.json.jinja:1]: I used datastacks to generate the gold example and this fix in the generated file was needed to deploy successfully

File: de_workloads/processing/gold_movies_example/spark_jobs/process.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    logger.info(f"Reading data from {SILVER_CONTAINER} container...")
+    ratings = read_data("ratings_small")
+    metadata = read_data("movies_metadata")
+
+    logger.info("Transforming data...")
+    avg_ratings = average_ratings(ratings)
+    output_df = join_metadata_and_ratings(metadata, avg_ratings)
+
+    logger.info(f"Saving data to {GOLD_CONTAINER} container...")
+    output_filepath = get_adls_file_url(GOLD_CONTAINER, "movies")
```
Comment (adurkan-amido) [inline de_workloads/processing/gold_movies_example/spark_jobs/process.py:72]: We should use the `OUTPUT_PATH_PATTERN` above, and write the table to something like `movies/movie_ratings_agg`

File: de_workloads/processing/gold_movies_example/spark_jobs/process.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            datasource_type=SOURCE_DATA_TYPE,
+        )
+
+    logger.info(f"Reading data from {SILVER_CONTAINER} container...")
+    ratings = read_data("ratings_small")
+    metadata = read_data("movies_metadata")
+
+    logger.info("Transforming data...")
+    avg_ratings = average_ratings(ratings)
+    output_df = join_metadata_and_ratings(metadata, avg_ratings)
```
Comment (adurkan-amido) [inline de_workloads/processing/gold_movies_example/spark_jobs/process.py:69]: As these 2 functions are one liners, I think we should just combine into one transformation function.
Also looking at the output table - it's all a bit contrived as they're only examples, but perhaps the drop the `belongs_to_collection` column, as wouldn't want something nested in the gold table.

File: docs/workloads/azure/data/etl_pipelines/data_processing.md [hunk 3]
```diff
@@ -45,7 +46,7 @@ Within Data Factory, the processing pipelines are kept within the `Process` fold
 
 ![ADF_SilverPipelinesList.png](../images/ADF_SilverPipelinesList.png)
 
-In Data Factory a standard data processing pipeline is as simple as this:
+In Data Factory a standard data processing pipeline is as simple as this example:
```
Comment (adurkan-amido) [inline docs/workloads/azure/data/etl_pipelines/data_processing.md:49]:
```suggestion
In Data Factory a data processing pipeline can be as simple as this example:
```

File: de_workloads/processing/gold_movies_example/README.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Data Processing Template Workload
```
Comment (adurkan-amido) [inline de_workloads/processing/gold_movies_example/README.md:1]: Might be good to update this readme with a brief overview of the workload (and we should do the same for the others).

**Review by adurkan-amido — CHANGES_REQUESTED** [review]
Looks good - just a few suggestions.

File: docs/workloads/azure/data/etl_pipelines/data_processing.md
Comment (adurkan-amido) [inline docs/workloads/azure/data/etl_pipelines/data_processing.md:1]: As a reminder - these updates will also need applying in the main docs repo.

File: de_workloads/processing/gold_movies_example/spark_jobs/process.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+    spark = get_spark_session_for_adls(WORKLOAD_NAME)
+
+    def read_data(table_name: str) -> DataFrame:
+        table_url = get_adls_file_url(SILVER_CONTAINER, INPUT_PATH_PATTERN.format(table_name=table_name))
+        return read_datasource(
+            spark,
+            table_url,
+            datasource_type=SOURCE_DATA_TYPE,
+        )
```
[the remark this answers is not in the document]
Comment (james-horrocks) [reply de_workloads/processing/gold_movies_example/spark_jobs/process.py:61]: We certainly could and I'm happy to do that.
My original thinking was that this saved repetitive code when loading from many tables, but I only ended up loading from 2 so it makes less sense now

**Review by adurkan-amido — APPROVED** [review]

**Pull request merged**